src/animation/BezierCurve.cpp

```diff
@@ -0,0 +1,88 @@
+#include <hyprutils/animation/BezierCurve.hpp>
+
+#include <array>
+#include <cmath>
+
+using namespace Hyprutils::Animation;
+using namespace Hyprutils::Math;
+
+void CBezierCurve::setup(const std::array<Vector2D, 2>& pVec) {
+    //const auto BEGIN = std::chrono::high_resolution_clock::now();
+
+    // Avoid reallocations by reserving enough memory upfront
+    m_vPoints.resize(pVec.size() + 2);
+    m_vPoints = {
+        Vector2D(0, 0),   // Start point
+        pVec[0], pVec[1], // Control points
+        Vector2D(1, 1)    // End point
+    };
+
+    if (m_vPoints.size() != 4)
+        std::abort();
+
+    // bake BAKEDPOINTS points for faster lookups
+    // T -> X ( / BAKEDPOINTS )
+    for (int i = 0; i < BAKEDPOINTS; ++i) {
+        float const t     = (i + 1) / (float)BAKEDPOINTS;
+        m_aPointsBaked[i] = Vector2D(getXForT(t), getYForT(t));
+    }
+
+    //const auto ELAPSEDUS  = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::high_resolution_clock::now() - BEGIN).count() / 1000.f;
+    //const auto POINTSSIZE = m_aPointsBaked.size() * sizeof(m_aPointsBaked[0]) / 1000.f;
+
+    //const auto BEGINCALC = std::chrono::high_resolution_clock::now();
+    for (int j = 1; j < 10; ++j) {
+        float i = j / 10.0f;
+        getYForPoint(i);
+    }
+    //const auto ELAPSEDCALCAVG = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::high_resolution_clock::now() - BEGINCALC).count() / 1000.f / 10.f;
+
+    //Debug::log(LOG, "Created a bezier curve, baked {} points, mem usage: {:.2f}kB, time to bake: {:.2f}µs. Estimated average calc time: {:.2f}µs.", BAKEDPOINTS, POINTSSIZE,
+    //           ELAPSEDUS, ELAPSEDCALCAVG);
```

Comment:
if you comment this out might as well remove it

Reply:
Yeah left them so i don't forget to ask if we want to keep them somehow? There are no logging facilities in hyprutils, so i just commented them out for now. We could return those metrics and log it in hyprland.

Reply:
nope, can be removed. It was for testing back 2 years ago.